spec/forms/bet_form_spec.rb

```diff
@@ -2,84 +2,85 @@
+          football_match_id: match.id,
+          home_team_score: -3,
+          away_team_score: -1
+        )
 
-      expect { form.save }.to change { Bet.count }.by 1
-    end
-  end
+        expect { form.save }.not_to change { Bet.count }
+        expect(form.errors).to have_key(:home_team_score)
+        expect(form.errors).to have_key(:away_team_score)
+      end
 
-  it 'requires scores to be positive' do
-    user = create(:user)
-    match = create(:football_match)
+      it 'requires football_match_id' do
+        user = create(:user)
 
-    form = BetForm.new(
-      Bet.new,
-      user_id: user.id,
-      football_match_id: match.id,
-      home_team_score: -3,
-      away_team_score: -1
-    )
-
-    expect { form.save }.to change { Bet.count }.by 0
-    expect(form.errors).to have_key(:home_team_score)
-    expect(form.errors).to have_key(:away_team_score)
-  end
+        form = BetForm.new(
+          Bet.new,
+          user_id: user.id,
+          football_match_id: nil,
+          home_team_score: 4,
+          away_team_score: 1
+        )
 
-  it 'requires football_match_id' do
-    user = create(:user)
+        expect { form.save }.not_to change { Bet.count }
+        expect(form.errors).to have_key(:football_match_id)
+      end
 
-    form = BetForm.new(
-      Bet.new,
-      user_id: user.id,
-      football_match_id: nil,
-      home_team_score: 4,
-      away_team_score: 1
-    )
+      it 'requires user_id' do
+        match = create(:football_match)
 
-    expect { form.save }.to change { Bet.count }.by 0
-    expect(form.errors).to have_key(:football_match_id)
-  end
+        form = BetForm.new(
+          Bet.new,
+          user_id: nil,
+          football_match_id: match.id,
+          home_team_score: 4,
+          away_team_score: 1
+        )
 
-  it 'requires user_id' do
-    match = create(:football_match)
+        expect { form.save }.not_to change { Bet.count }
+        expect(form.errors).to have_key(:user_id)
+      end
 
-    form = BetForm.new(
-      Bet.new,
-      user_id: nil,
-      football_match_id: match.id,
-      home_team_score: 4,
-      away_team_score: 1
-    )
+      it "user can't bet on the same match twice " do
+        user = create(:user)
+        match = create(:football_match)
+        create(:bet, user_id: user.id, football_match_id: match.id)
 
-    expect { form.save }.to change { Bet.count }.by 0
-    expect(form.errors).to have_key(:user_id)
-  end
+        form = BetForm.new(
+          Bet.new,
+          user_id: user.id,
+          football_match_id: match.id,
+          home_team_score: 4,
+          away_team_score: 1
+        )
 
-  context 'when passed data is valid' do
-    it "user can't bet on the same match twice " do
-      user = create(:user)
-      match = create(:football_match)
-      create(:bet, user_id: user.id, football_match_id: match.id)
+        expect { form.save }.not_to(change { Bet.count })
+      end
+    end
+    context '#save' do
```

Comment:
Please add one line of separation.

Also, it is the place when we should check if a bet is saved with correct attributes.

Controller -> we would like to know that BetForm receives a call with correct attributes.
Form -> we would like to know that Bet is saved with correct attributes.

The whole point of including form objects is to separate responsibility from a controller to form objects (when we should test all validations and object creation).